annotations/views.py

```diff
@@ -218,6 +218,33 @@ def list_texts(request):
     }
     return HttpResponse(template.render(context))
 
+def list_users(request):
+    """
+    List all the users of Vogon web
+    """
+    queryset = VogonUser.objects.all()
+    template = loader.get_template('annotations/contributors.html')
+    userList = []
+    # text_list = Text.objects.all()
+    for user in queryset:
```

Comment:
@ndondeti @allaparthi92 This is a great start! Here on line 229, however, is one potential problem.

In Django, we construct and manipulate [QuerySets](https://docs.djangoproject.com/en/1.9/ref/models/querysets/) in advance of any database calls, and then when we're good and ready (e.g. when we want to render the template, or do something fun with the results) we evaluate the QuerySet and hit the database. This is great, because it lets us set up sequential filtering and such without asking the database for fifty bajillion rows; Django sets up the query logic and then (almost always) farms out the heavy lifting to the database backend (Postgres, in this case). 

Pagination is part of the process of whittling down the QuerySet to a manageable size, not just for the user but for our application as well! If we have 10,000 users, we don't want to load all 10,000 rows from the user table and then just display 25 of them. It's better to get the 25 that we want. So we use Paginator to refine our QuerySet before it gets evaluated.

**BUT** if you iterative over a QuerySet, it will be evaluated. That means that, right here on line 229, you're making a database call with the QuerySet created on line 225: *all* of the VogonUsers. I.e. ``SELECT * FROM TABLE annotations_vogonuser;``. Yikes! And then you're passing to the Paginator not a QuerySet, but a list of VogonUser instances. The Paginator won't complain, but your users will when they have to wait five minutes for the page to load.

Take a look at the excellent [QuerySet documentation](https://docs.djangoproject.com/en/1.9/ref/models/querysets/) for more information.